Microsoft.Toolkit.Uwp.UI/Brushes/RadialGradientBrush.cs

```diff
@@ -0,0 +1,115 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+//// UWP Replacement for WPF RadialGradientBrush: https://msdn.microsoft.com/en-us/library/system.windows.media.radialgradientbrush(v=vs.110).aspx.
+
+using Microsoft.Graphics.Canvas;
+using Microsoft.Graphics.Canvas.Brushes;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using System.Numerics;
+using Windows.UI;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Brushes
+{
+    /// <summary>
+    /// RadialGradientBrush - This GradientBrush defines its Gradient as an interpolation
+    /// within an Ellipse.
+    /// </summary>
+    [ContentProperty(Name = nameof(GradientStops))]
+    public partial class RadialGradientBrush : Win2DCanvasBrushBase
+    {
+        private static void OnPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var brush = (RadialGradientBrush)d;
+
+            // We need to recreate the brush on any property change.
+            brush.OnDisconnected();
+            brush.OnConnected();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RadialGradientBrush"/> class.
+        /// </summary>
+        public RadialGradientBrush()
+        {
+            // Rendering surface size, if this is too small the gradient will be pixelated.
+            // This seems like a good compromise.
+            this.SURFACE_RESOLUTION_X = 512;
```

Comment:
Hardcoding this, is this affect by PPI? Would 4K or 8K or future resolutions be affected by this?

Reply:
also `this` is not necessary here

Reply:
@WilliamABradley it does slightly, but it's not as noticeable in most cases unless you have a very noisy gradient.  This ties back into us not knowing how to get the target visual's size yet (if at all possible).  I think in the future based on reception and what we figure out is we'll either expose it as a property, or we'll use an attached property on the parent UIElement to get the size, which we can use here and for an Absolute MappingMode.